core/platform/datastore/gae_datastore_services_test.py

```diff
@@ -90,11 +97,13 @@ class MockDatetimeForDatastoreTests(test_utils.GenericTestBase):
     """Tests for mocking the datetime of an ndb.Model."""
 
     def test_exception_is_raised_when_passed_wrong_type(self):
-        with self.assertRaisesRegexp(Exception, 'mocked_now must be datetime'):
-            with gae_datastore_services.mock_datetime_for_datastore(2020):
+        # type: () -> None
+        with self.assertRaisesRegexp(Exception, 'mocked_now must be datetime'): # type: ignore[no-untyped-call]
+            with gae_datastore_services.mock_datetime_for_datastore(2020): # type: ignore[arg-type]
```

Comment:
@seanlip There are some ignores which I think must be used:
1. no-untyped-call: Already approved to use due to partially typed codebase.
2. arg-type: In test files, we try to provide wrong type values intentionally to check how the function behaves. So this should be used. Example is this case itself.
3. [override](https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides): For example see [delete](https://github.com/oppia/oppia/blob/3d808927f2984e2e92dfbc0ba9b3cbcdf4c1ad3e/core/storage/base_model/gae_models.py#L342) of base_model and [delete](https://github.com/oppia/oppia/blob/3d808927f2984e2e92dfbc0ba9b3cbcdf4c1ad3e/core/storage/base_model/gae_models.py#L836) of VersionedModel. Here mypy gives error that signature of VersionedModel.delete is incompatible with that of BaseModel.delete. I have used this ignore 12 times in core/storage and core/platform till now. 

Here we have a couple of options for point 2 and 3:
- Add a comment above every ignore.
- Add a point in the wiki about using this.
   What do you think is most appropriate?

Reply:
For override, can I change the signature of BaseModel.delete? I think I'll try this

Reply:
For arg type, @sajalasati had a note about dropping tests for invalid types if mypy will already check this. I'm fine with whatever you decide (to drop those tests now or later). If you drop them later, though, then every occurrence of arg-type should have a comment explaining why and a TODO pointing to the issue, so that it is easy to find all the incidents later on when we do a sweep and delete them.

For override, I thought about it and the decision on what to do here is tricky. I would suggest changing the signature of BaseModel.delete to accept *args, and putting a comment above that to say "The unused *args here is needed in order to support additional arguments like committer_id in subclasses like VersionedModel while still keeping the method signatures consistent." Would that work?

For the general things, please do both: add a comment above every ignore (except point 1) **and** add a point in the wiki as well. For 2 and 3, make the comments specific to each case rather than a boilerplate comment that is repeated everywhere.

Reply:
I tried adding *args to the BaseModel.delete. Still mypy says that Signature of delete is not compatible with the base model. I think we should continue to use override in this case as adding *args is not working.

Reply:
In this specific case, I think it is fine to use override, but please include a clear comment with explanation for why we are doing this. To double-check, is it correct that "override" only needs to be annotated just once (rather than at every callsite)?

Please also note that this does not constitute a general exception for other cases in the codebase, and other usages of override are not approved by default.

Reply:
> To double-check, is it correct that "override" only needs to be annotated just once (rather than at every callsite)?

Yes, this ignore is only needed when function is declared. Not when function is called.

> Please also note that this does not constitute a general exception for other cases in the codebase, and other usages of override are not approved by default.

There are 4 similar cases:
1. [_Gae.import_models() ](https://github.com/oppia/oppia/blob/35bdbe6d101c2033ceb4b738a78f6f4502bb08eb/core/platform/models.py#L72) with [Platform.import_models()](https://github.com/oppia/oppia/blob/35bdbe6d101c2033ceb4b738a78f6f4502bb08eb/core/platform/models.py#L50)
2. [VersionedModel.delete_multi()](https://github.com/oppia/oppia/blob/35bdbe6d101c2033ceb4b738a78f6f4502bb08eb/core/storage/base_model/gae_models.py#L897) and [CollectionModel.delete_multi()](https://github.com/oppia/oppia/blob/35bdbe6d101c2033ceb4b738a78f6f4502bb08eb/core/storage/collection/gae_models.py#L290) with [BaseModel.delete_multi()](https://github.com/oppia/oppia/blob/35bdbe6d101c2033ceb4b738a78f6f4502bb08eb/core/storage/base_model/gae_models.py#L324)
3. [ConfigPropertyModel.commit()](https://github.com/oppia/oppia/blob/35bdbe6d101c2033ceb4b738a78f6f4502bb08eb/core/storage/config/gae_models.py#L73) with[ VersionedModel.commit()](https://github.com/oppia/oppia/blob/35bdbe6d101c2033ceb4b738a78f6f4502bb08eb/core/storage/base_model/gae_models.py#L998)